src/jace/util/compiling.py

```diff
@@ -0,0 +1,146 @@
+    For running the SDFG you also have to pass the input names (`inp_names`) and output names
+    (`out_names`) that were inside the `TranslatedJaxprSDFG` from which `csdfg` was compiled from.
+
+    Args:
+        csdfg:      The `CompiledSDFG` object.
+        inp_names:  List of names of the input arguments.
+        out_names:  List of names of the output arguments.
+        cargs:      All positional arguments of the call.
+        ckwargs:    All keyword arguments of the call.
+
+    Note:
+        There is no pytree mechanism jet, thus the return values are returned inside a `tuple`
+        or in case of one value, directly, in the order determined by Jax.
+        Furthermore, DaCe does not support scalar return values, thus they are silently converted
+        into arrays of length 1, the same holds for inputs.
+
+    Todo:
+        - Since we do not have symbols and a fixed size this works and there is no problem.
+            However, if we have symbols or variable sizes, we must ensure that the init function of
+            the SDFG is called every time, or ensure that its exit function runs every time.
+        - Implement non C strides.
+    """
+    sdfg: dace.SDFG = csdfg.sdfg
+
+    if len(ckwargs) != 0:
+        raise NotImplementedError("No kwargs are supported yet.")
+    if len(inp_names) != len(cargs):
+        raise RuntimeError("Wrong number of arguments.")
+    if len(sdfg.free_symbols) != 0:  # This is a simplification that makes our life simple.
+        raise NotImplementedError(
+            f"No externally defined symbols are allowed, found: {sdfg.free_symbols}"
+        )
+
+    # Build the argument list that we will pass to the compiled object.
+    call_args: dict[str, Any] = {}
+    for in_name, in_val in zip(inp_names, cargs, strict=True):
+        if util.is_scalar(in_val):
+            # Currently the translator makes scalar into arrays, this has to be reflected here
+            in_val = np.array([in_val])
+        call_args[in_name] = in_val
+
+    for out_name, sarray in ((name, sdfg.arrays[name]) for name in out_names):
+        if out_name in call_args:
+            if util.is_jax_array(call_args[out_name]):
+                # Jax arrays are immutable, so they can not be return values too.
+                raise ValueError("Passed a Jax array as output.")
+        else:
+            call_args[out_name] = dace_data.make_array_from_descriptor(sarray)
+
+    assert len(call_args) == len(csdfg.argnames), (
+        "Failed to construct the call arguments,"
+        f" expected {len(csdfg.argnames)} but got {len(call_args)}."
+        f"\nExpected: {csdfg.argnames}\nGot: {list(call_args.keys())}"
+    )
+
+    # Calling the SDFG
+    with dace.config.temporary_config():
+        dace.Config.set("compiler", "allow_view_arguments", value=True)
+        csdfg(**call_args)
+
+    # Handling the output (pytrees are missing)
+    if len(out_names) == 0:
+        return None
```

Comment:
This is contradicting the return type annotations. Is the type checker complaining here? If not, why?

Reply:
MyPy does not complain.